node/pkg/telemetry/loki.go

```diff
@@ -142,15 +163,133 @@ func NewLokiCloudLogger(ctx context.Context, logger *zap.Logger, url string, pro
 		lokiLabels[level] = levLabels
 	}
 
+	// Create a buffered channel so the application does not block in the logger.
+	bufferedChan := make(chan api.Entry, bufferSize)
+
+	// Create a local context with a cancel function so we can signal our worker to shutdown when the time comes.
```

Comment:
Maybe also "Cancelling the worker will also stop the connection to the Loki client."